weave/mongo_loader.py

```diff
@@ -64,7 +65,7 @@ def __init__(self, pantry, mongo_client=None, **kwargs):
         # Get the database. (Use MONGODB_DATABASE, defaulting to
         # pantry_path if it is not present.)
         self.database_name = self.mongo_config.get(
-            "mongodb_database", self.pantry.pantry_path)
```

Comment:
I see what your intention is.

There is a bug when pantry path points to a non root dir, ie (data/subfolder1/LocalPantry) due to the path separators in the pantry_path attribute.
However, I don't think this solution solves this issue entirely. 
Path.name will get the last folder name, which could potentially lead to two different pantries using the same mongodb. For example, if we had two local pantries in the following locations, they would use the same mongo db despite tracking two different directories:
data/subfolder1/LocalPantry
data/subfolder2/LocalPantry

Path(data/subfolder*/LocalPantry).name would return "LocalPantry" in both cases.

We have the same problem with the SQL database tracking. We used the following solution to keep the database names unique while still using the pantry path as the basis:
```self.database_name = pantry_path.replace(os.sep, "_") .replace("-", "_")```

For mongo, you might need to make some adjustments based on their database name requirements (sql required underscores instead of dashes, mongo might have some other requirements), but doing it in this way will prevent future problems with pantries that have the same final folder name.

Reply:
Makes sense, I'll look into that

Reply:
Ok, this makes sense, though I found that for the purposes of the demo, this change isn't needed. 

My suggestion is I cancel this merge request and we make a separate ticket to address this.